modules/utxo-lib/test/bitgo/psbt/psbtUtil.ts

```diff
@@ -0,0 +1,278 @@
+  }
+}
+
+export function validatePsbtParsing(
+  tx: UtxoTransaction<bigint>,
+  psbt: UtxoPsbt<UtxoTransaction<bigint>>,
+  unspents: WalletUnspent<bigint>[],
+  signatureTarget: SignatureTargetType
+): void {
+  unspents.forEach((u, i) => {
+    if (isWalletUnspent(u)) {
+      const sciptType = scriptTypeForChain(u.chain);
+      const psbtParsed = parsePsbtInput(psbt, i);
+
+      if (signatureTarget === 'unsigned') {
+        if (sciptType === 'p2tr') {
+          assert.deepStrictEqual(psbtParsed, undefined);
+        } else {
+          assert.ok(psbtParsed);
+          assert.deepStrictEqual(psbtParsed?.scriptType, sciptType);
+          validateScript(psbtParsed, undefined);
+          validatePublicKeys(psbtParsed, undefined);
+          validateSignature(psbtParsed, undefined);
+        }
+      } else {
+        assert.ok(psbtParsed);
+        const txParsed = parseSignatureScript2Of3(tx.ins[i]);
+        assert.deepStrictEqual(psbtParsed?.scriptType, sciptType);
+        validateScript(psbtParsed, txParsed);
+        validatePublicKeys(psbtParsed, txParsed);
+        validateSignature(psbtParsed, txParsed);
+      }
+    }
+  });
+}
+
+export function toBigInt<TNumber extends number | bigint>(unspents: Unspent<TNumber>[]): WalletUnspent<bigint>[] {
+  return unspents.map((u) => {
+    if (isWalletUnspent(u)) {
+      return { ...u, value: BigInt(u.value) };
+    } else {
+      throw new Error(`invalid unspent`);
+    }
+  });
+}
+
+export function signPsbt(
+  psbt: UtxoPsbt<UtxoTransaction<bigint>>,
+  unspents: Unspent<bigint>[],
+  rootWalletKeys: RootWalletKeys,
+  signer: string,
+  cosigner: string,
+  signatureTarget: SignatureTargetType
+): void {
+  unspents.forEach((u, i) => {
+    if (isWalletUnspent(u)) {
+      try {
+        if (signatureTarget === 'unsigned') {
+          signWalletPsbt(psbt, i, rootWalletKeys[signer], u);
+        }
+        signWalletPsbt(psbt, i, rootWalletKeys[cosigner], u);
+
+        // unsigned p2tr does not contain tapLeafScript & tapBip32Derivation.
+        // So it's signing is expected to fail
+        assert.deepStrictEqual(
+          signatureTarget === 'unsigned' && scriptTypeForChain(u.chain) === 'p2tr',
+          false,
+          'unsigned p2tr sign should fail'
+        );
+      } catch (err) {
+        if (err.message === 'unsigned p2tr sign should fail') {
+          throw err;
+        }
+        assert.deepStrictEqual(signatureTarget, 'unsigned');
+        assert.deepStrictEqual(scriptTypeForChain(u.chain), 'p2tr');
+        assert.deepStrictEqual(psbt.data.inputs[i].tapLeafScript, undefined);
+        assert.deepStrictEqual(psbt.data.inputs[i].tapBip32Derivation, undefined);
+        assert.deepStrictEqual(psbt.data.inputs[i].tapScriptSig, undefined);
+        assert.ok(psbt.data.inputs[i].witnessUtxo);
+      }
+    } else {
+      throw new Error(`invalid unspent`);
+    }
+  });
+}
+
+export function signTxBuilder<TNumber extends number | bigint>(
```

Comment:
Looks very similar to some methods in `test/transaction_util.ts`. Please check if you can consolidate.

Reply:
Not straight forward. Keeping it for now.